src/main/resources/static/common/js/demoAdmin.js

```diff
@@ -753,21 +753,32 @@ mod.controller('demo_admin_controller', function($rootScope, $scope, $http, $fil
 	    }
 
 		function addNewItem(){
-			$http({
-	            method: 'POST',
-	            url: '/proxy/v1/assets/items/',
-	            data: angular.element('#item_form').serializeJSON(),
-	            headers : {'Content-Type': 'application/json'}
-	        }).then(function(response) {
-	        	 toastr.success($filter('translate')("ADD_ITEM_SUCCESS"));
-	        	 demo.itemModal.showErrorBox = false;
-	        	 getAllItems();
-	        	 $('#item_modal').modal('hide');
-	        }, function error(response) {
-	        	handleErrorMessageForItemEdit(response);
-	        }).catch(function(response) {
-	        	handleErrorMessageForItemEdit(response);
-	        });
+            let params = angular.element('#item_form').serializeJSON();
+            let success = (data) => {
+                toastr.success($filter('translate')("ADD_ITEM_SUCCESS"));
+                demo.itemModal.showErrorBox = false;
+                getAllItems();
+                $('#item_modal').modal('hide');
+            }
+            let failToAddNewItem = (data) => {
+                handleErrorMessageForItemEdit(data);
+            }
+            if (CURRENT_WEB_SERVICE_MODE === "GraphQL") {
+                graphQLService.addNewItem(params, success, (data) => {failToAddNewItem(data)}, "{id}");
+            } else {
+                $http({
+                    method: 'POST',
+                    url: '/proxy/v1/assets/items/',
+                    data: params,
+                    headers : {'Content-Type': 'application/json'}
+                }).then(function(response) {
+                    success(response);
+                }, function error(response) {
+                    failToAddNewItem(response);
```

Comment:
Just put `handleErrorMessageForItemEdit` method here. No need to create `failToAddNewItem`